main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -450,16 +450,16 @@ object Typer extends Phase[KindedAst.Root, TypedAst.Root] {
         for {
           (constrs, bodyType, bodyEff) <- visitExp(exp)
           _ <- unifyTypeM(argType, argTypeVar, loc)
-          resultTyp <- unifyTypeM(tvar, Type.mkArrowWithEffect(argType, bodyEff, bodyType), loc)
+          resultTyp <- unifyTypeM(tvar, Type.mkArrowWithEffect(argType, bodyEff, bodyType, loc), loc)
         } yield (constrs, resultTyp, Type.Pure)
 
       case KindedAst.Expression.Apply(exp, exps, tvar, evar, loc) =>
-        val lambdaBodyType = Type.freshVar(Kind.Star)
-        val lambdaBodyEff = Type.freshVar(Kind.Bool)
+        val lambdaBodyType = Type.freshVar(Kind.Star, loc = loc)
```

Comment:
Whats the reason for the use of named parameters suddenly? I don't think `loc = loc` adds much value.

Reply:
Type.freshVar has optional parameters. `loc` is not optional but it comes after the optional ones so the named param style is required. I'll move it ahead of the optional params.